### PR TITLE
Use Docker Compose v2

### LINKED ISSUE #119: Support for `docker compose` instead of `docker-compose`
This should be a really simple fix, by maybe adding a try catch logic at `bin/docker-compose`.

### PR DESCRIPTION
**Pull request by emcsween**
## Description

This makes Docker Compose v2 the default, with a fallback to Docker Compose v1 if v2 is unavailable.

The bin/docker-compose script was getting big, so I refactored it into smaller functions organized around the different optional docker-compose config files. It will be easier to review this PR commit by commit.

While I was refactoring, I noticed that the REDIS_PORT config variable was not propagated to the sharelatex container, so I fixed that.

## Related issues / Pull Requests

- Fixes #119 
- Implements #124 in a backwards compatible way

## Manual testing

I tested `bin/up` and `bin/doctor` with the following configurations:

* docker-compose-plugin (v2) installed, docker-compose (v1) installed
* docker-compose-plugin (v2) installed, docker-compose (v1) uninstalled
* docker-compose-plugin (v2) uninstalled, docker-compose (v1) installed
* docker-compose-plugin (v2) uninstalled, docker-compose (v1) uninstalled

### CODE REVIEW

**Comment by mserranom** [comment]
Since we're here, there are also references and a log output for docker-compose in https://github.com/overleaf/toolkit/blob/master/doc/the-doctor.md.

File: doc/quick-start-guide.md
Comment (mserranom) [inline doc/quick-start-guide.md:165]: From the changes in `doctor` we're still presenting a `docker compose` entry (just removing the hyphen)

Comment (mserranom) [reply doc/quick-start-guide.md:165]: Although if there's no `docker compose` it will check for  `docker-compose`. I guess just undoing this deletion is just fine (even keeping it is OK, we just want to send the literal report)

**Review by mserranom — APPROVED** [review]
Looks good, left a couple of very minor comments. The refactor of bin/docker-compose is brilliant, much more readable now 🎉

**Comment by emcsween** [comment]
@mserranom, I realized that the bin/logs script didn't properly clean up background jobs with compose v2. Killing a `docker compose exec` process in v2 doesn't kill the process the command started inside the container. I had to work around this by recording the PIDs inside the container and killing them from inside the container at the end. This is in https://github.com/overleaf/toolkit/pull/158/commits/ae03cd35683181b5129e12a22b0b16a4ae91131d. I'll let you review that change as well before I merge.

I tested `bin/logs` and `bin/logs  -f` with both Docker Compose v1 and v2, and in the following states:

* No running services
* Only mongo running
* All services running

**Review by mserranom — APPROVED** [review]

**Pull request merged**
